### ISSUE TITLE
'can not access a member' exception

### ISSUE THREAD

**Issue by idan-ratzabi** [fabioCollini/DaggerMock#55]
My setup is :

```
public class RoboElectricDaggerMockRule extends DaggerMockRule<TestAppComponent> {
    public RoboElectricDaggerMockRule() {
        super(TestAppComponent.class, new AssistantUtilsModule());
        customizeBuilder(new BuilderCustomizer<TestAppComponent.Builder>() {
            @Override public TestAppComponent.Builder customize(TestAppComponent.Builder builder) {
                return builder.application(getApp());
            }
        });
        set(new DaggerMockRule.ComponentSetter<TestAppComponent>() {
            @Override public void setComponent(TestAppComponent component) {
                component.inject(getApp());
            }
        });
    }

    private static AnydoRobolectricTestApp getApp() {
        return (AnydoRobolectricTestApp) RuntimeEnvironment.application;
    }
}
```
My test configuration :

```
@RunWith(RobolectricTestRunner.class)
@Config(application = AnydoRobolectricTestApp.class,constants = BuildConfig.class, sdk = 21, shadows = { ShadowPreferenceHelper.class})
public class DeepLinkTestNew {


    @Rule public RoboElectricDaggerMockRule rule = new RoboElectricDaggerMockRule();

    @Mock
    AssistantUtils assistantUtils;
    
    @Test
    `public void testSomething()`
```
```

public interface TestAppComponent extends AndroidInjector<AnydoRobolectricTestApp> {

    @Override
    void inject(AnydoRobolectricTestApp instance);

    @Component.Builder
    interface Builder {

        Builder assistantUtilsModule(AssistantUtilsModule assistantUtilsModule);

        @BindsInstance
        TestAppComponent.Builder application(Application application);

        TestAppComponent build();
    }
}
```

But I get the following error :
![image](https://user-images.githubusercontent.com/2378505/30278104-430eac42-9712-11e7-9702-59cf18005600.png)

> "java.lang.IllegalAccessException: Class it.cosenonjaviste.daggermock.ObjectWrapper can not access a member of class com.anydo.di.components.DaggerTestAppComponent$Builder with modifiers "public""

TestAppComponent which is auto generated by Dagger has all fields private, so it makes sense. 
Am I missing something ?

And another question, is it mandatory to add a module setter like this inside the component?
`  Builder assistantUtilsModule(AssistantUtilsModule assistantUtilsModule);`
it's kinda boilerplate and would be great if we can remove.

**Comment by fabioCollini** [comment]
Hi, which Dagger version are you using? Are you using Dagger Android or `BindInstance` annotation? Can you try to use DaggerMock [0.7.1-beta1](https://jitpack.io/#fabioCollini/DaggerMock/0.7.1-beta1)?
Thanks for your report

**Comment by thebestpol** [comment]
This problem continues with the latest version 0.8.1. Someone have a solution?

**Comment by fabioCollini** [comment]
Hi, could you please share the complete stack trace of the error? Are you using a robolectric test?
Thanks for your report

**Comment by thebestpol** [comment]
The problem is using Robolectric test runner with a custom TestRule, the same as @idan-ratzabi 

```
public class RobolectricMockTestRule extends DaggerMockRule<AppComponent> {

    public RobolectricMockTestRule() {
        super(AppComponent.class, new AppModule(getApplication()));

        customizeBuilder(new BuilderCustomizer<AppComponent.Builder>() {
            @Override
            public AppComponent.Builder customize(AppComponent.Builder builder) {
                return builder.application(getApplication());
            }
        });

        set(new ComponentSetter<AppComponent>() {
            @Override
            public void setComponent(AppComponent component) {
                component.inject(getApplication());
            }
        });
    }

    private static SheddApplication getApplication() {
        return ((SheddApplication) RuntimeEnvironment.application);
    }
}
```

Using this dagger Component

```
@Singleton
@Component(modules = {
        AndroidSupportInjectionModule.class,
        ActivityBindingModule.class,
        AppModule.class
})
public interface AppComponent {
    @Component.Builder
    interface Builder {
        Builder appModule(AppModule appModule);

        @BindsInstance
        Builder application(SheddApplication app);

        AppComponent build();
    }

    void inject(SheddApplication app);
}
```

And the error stack trace:

```
java.lang.RuntimeException: Error invoking setter with parameter class com.sheddapp.di.AppModule on object com.sheddapp.di.DaggerAppComponent$Builder@514cd540

	at it.cosenonjaviste.daggermock.ObjectWrapper.invokeBuilderSetter(ObjectWrapper.java:173)
	at it.cosenonjaviste.daggermock.DaggerMockRule.initComponent(DaggerMockRule.java:238)
	at it.cosenonjaviste.daggermock.DaggerMockRule.setupComponent(DaggerMockRule.java:130)
	at it.cosenonjaviste.daggermock.DaggerMockRule.access$000(DaggerMockRule.java:36)
	at it.cosenonjaviste.daggermock.DaggerMockRule$1.evaluate(DaggerMockRule.java:110)
	at org.robolectric.internal.SandboxTestRunner$2.evaluate(SandboxTestRunner.java:228)
	at org.robolectric.internal.SandboxTestRunner.runChild(SandboxTestRunner.java:110)
	at org.robolectric.internal.SandboxTestRunner.runChild(SandboxTestRunner.java:37)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.robolectric.internal.SandboxTestRunner$1.evaluate(SandboxTestRunner.java:64)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:117)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:42)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:262)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:84)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)
Caused by: java.lang.IllegalAccessException: Class it.cosenonjaviste.daggermock.ObjectWrapper can not access a member of class com.sheddapp.di.DaggerAppComponent$Builder with modifiers "public"
	at sun.reflect.Reflection.ensureMemberAccess(Reflection.java:102)
	at java.lang.reflect.AccessibleObject.slowCheckMemberAccess(AccessibleObject.java:296)
	at java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:288)
	at java.lang.reflect.Method.invoke(Method.java:491)
	at it.cosenonjaviste.daggermock.ObjectWrapper.invokeBuilderSetter(ObjectWrapper.java:171)
	... 24 more
```

Is it possible to use AndroidInjector and DaggerMock in Robolectric tests?

**Comment by fabioCollini** [comment]
I have pushed a new version to fix this issue, I have tried [here](https://github.com/fabioCollini/DaggerMock/tree/master/RealWorldAppInjector/src/test/java/it/cosenonjaviste/daggermock/robolectric) and it seems to work.
Can you try the version [cc0b0acd98](https://jitpack.io/#fabioCollini/DaggerMock/cc0b0acd98) to confirm that this commit fixes your error as well?
Thanks again for your help

**Comment by thebestpol** [comment]
@fabioCollini Thanks you!! It works as expected with the cc0b0acd98 build. 

I think you could close this issue, great job.

**Comment by fabioCollini** [comment]
I have just released version 0.8.2 to include this fix, thanks again for your report and for your help!
